pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -638,13 +647,78 @@ func (a *index) inRevision(obj any) bool {
 
 // All return all known workloads. Result is un-ordered
 func (a *index) All() []model.AddressInfo {
-	res := slices.Map(a.workloads.List(), modelWorkloadToAddressInfo)
+	var res []model.AddressInfo
+	// Add all workloads
+	for _, wl := range a.workloads.List() {
+		// If EnableAmbientMultiNetwork is enabled, we only want to add workloads that are local to this cluster
```

Comment:
Not necessarily right? The flag is for multinetwork, but this approach should ideally be compatible with single network so we'd want workloads from other clusters. Does it make sense to just add the workload's network to WorkloadInfo (but not the WDS)?

Reply:
also from other PR

Reply:
I thought Steven had added this or would add this, but I am happy to add it in my PR.

Reply:
https://github.com/istio/istio/pull/56844/files/a4512cb2adea47a8dc80b3e9dfb2d4e61550b68d#r2191090214